src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -7391,6 +7391,249 @@ namespace SimAirServingZones {
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // FUNCTION LOCAL VARIABLE DECLARATIONS:
+        // na
+
+        CheckWaterCoilIsOnAirLoop = CheckWaterCoilOnPrimaryAirLoopBranch(CompTypeNum, CompName);
+        if (!CheckWaterCoilIsOnAirLoop) {
+            CheckWaterCoilIsOnAirLoop = CheckWaterCoilOnOASystem(CompTypeNum, CompName);
+        }
+
+        if (!CheckWaterCoilIsOnAirLoop) {
+            CheckWaterCoilIsOnAirLoop = CheckWaterCoilSystemOnAirLoopOrOASystem(CompTypeNum, CompName);
+        }
+        if (!CheckWaterCoilIsOnAirLoop) {
+            ShowSevereError("CheckWaterCoilIsOnAirLoop: = " + CompType + " = " + CompName + ".");
+            ShowContinueError("The water coil or coil system is neither on primary air branch nor on outdoor air system hence does not require "
+                              "'Controller:WaterCoil' object.");
+        }
+
+        WaterCoilOnAirLoop = CheckWaterCoilIsOnAirLoop;
+    }
+
+    bool CheckWaterCoilOnPrimaryAirLoopBranch(int const CompTypeNum, std::string const CompName)
+    {
+
+        // PURPOSE OF THIS FUNCTION:
+        // This function returns true if a water coil that has water controller is on
+        // primary air loop branch. Searches for water coil name and type that match
+        // components list in primary air systems.
+
+        // METHODOLOGY EMPLOYED:
+        // na
+
+        // REFERENCES:
+        // na
+
+        // USE STATEMENTS:
+        // using HVACHXAssistedCoolingCoil::GetCoilsInputFlag;
+        // na
+
+        // Return value
+        bool CheckWaterCoilIsOnPrimaryAirLoopBranch(false);
+
+        // Locals
+        // FUNCTION ARGUMENT DEFINITIONS:
+
+        // FUNCTION PARAMETER DEFINITIONS:
+        // na
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // FUNCTION LOCAL VARIABLE DECLARATIONS:
+        // na
+
+        if (GetAirLoopInputFlag) { // First time subroutine has been entered
+            GetAirPathData();      // Get air loop descriptions from input file
+            GetAirLoopInputFlag = false;
+        }
+
+        CheckWaterCoilIsOnPrimaryAirLoopBranch = false;
+        if (DataHVACGlobals::NumPrimaryAirSys > 0) {
+            for (int AirSysNum = 1; AirSysNum <= DataHVACGlobals::NumPrimaryAirSys; ++AirSysNum) {
+                for (int BranchNum = 1; BranchNum <= PrimaryAirSystem(AirSysNum).NumBranches; ++BranchNum) {
+                    for (int CompNum = 1; CompNum <= PrimaryAirSystem(AirSysNum).Branch(BranchNum).TotalComponents; ++CompNum) {
+                        if ((CompTypeNum == PrimaryAirSystem(AirSysNum).Branch(BranchNum).Comp(CompNum).CompType_Num) &&
+                            UtilityRoutines::SameString(CompName, PrimaryAirSystem(AirSysNum).Branch(BranchNum).Comp(CompNum).Name)) {
+                            CheckWaterCoilIsOnPrimaryAirLoopBranch = true;
+                            goto PrimaryAirSystemLoop_exit;
```

Comment:
Annnd nope.  Just put a return statement in here.  Get rid of the local variable and **no goto**.

```
if (Sys > 0) {
  for each air system {
    for each branch {
      for each comp {
        if comp is type and same string name {
          return true;
        }
      }
    }
  }
}
return false;
```

Reply:
Implemented as suggested.